generator/src/test/java/com/scottlogic/deg/generator/cucumber/OfLength.feature

```diff
@@ -1,593 +1,455 @@
-    Given foo is of length <length>
-    And foo is in set:
-      | "a"          |
-      | "  10  10  " |
-    Then the following data should be generated:
-      | foo        |
-      | null       |
-      | <expected> |
-    Examples:
-      | length | expected     |
-      | 10     | "  10  10  " |
-
-
-  Scenario: Running an 'ofLength' request that includes a numeric value = 0 on an empty string should be successful
-    Given foo is of length 0
-    And foo is in set:
-      | ""  |
-      | "a" |
-    Then the following data should be generated:
-      | foo  |
-      | null |
-      | ""   |
+Background:
+     Given the generation strategy is full
+       And there is a field foo
+       And foo is of type "string"
+
+Scenario Outline: Running an 'ofLength' request on a roman alphabet character string should be successful
+     Given foo is of length <length>
+       And foo is in set:
+         | "a"  |
+         | "aa" |
+     Then the following data should be generated:
+       | foo        |
+       | null       |
+       | <expected> |
+       And the following data should not be included in what is generated:
+         | foo           |
+         | <notExpected> |
+     Examples:
+       | length | expected | notExpected |
+       | 1      | "a"      | "aa"        |
+       | 2      | "aa"     | "a"         |
+       | 1.0    | "a"      | "aa"        |
+
+Scenario Outline: Running an 'ofLength' request that includes a negation of a valid numeric length should be successful
+     Given foo is anything but of length <length>
+       And foo is in set:
+         | "a"  |
+         | "aa" |
+     Then the following data should be generated:
+       | foo        |
+       | null       |
+       | <expected> |
+       And the following data should not be included in what is generated:
+         | foo           |
+         | <notExpected> |
+     Examples:
+       | length | expected    | notExpected |
+       | 1      | "aa"        | "a"         |
+       | 2      | "a"         | "aa"        |
+
+Scenario Outline: Running an 'ofLength' request on a roman alphabet character string value and white space should be successful
+     Given foo is of length <length>
+       And foo is in set:
+         | "a"          |
+         | "  10  10  " |
+     Then the following data should be generated:
+       | foo        |
+       | null       |
+       | <expected> |
+     Examples:
+       | length | expected     |
+       | 10     | "  10  10  " |
+
+Scenario: Running an 'ofLength' for a length of zero should be successful
+     Given foo is of length 0
+       And foo is in set:
+         | ""  |
+         | "a" |
+     Then the following data should be generated:
+       | foo  |
+       | null |
+       | ""   |
+
+Scenario Outline: Running an 'ofLength' request that includes a decimal number containing non zero digits should fail with an error message
```

Comment:
The implication here is that `ofLength 1.00` would work, do we have a scenario for that?

Reply:
Yes it's in `Scenario Outline: Running an 'ofLength' request on a roman alphabet character string should be successful`

Reply:
True, I didn't spot that one